message.go

```diff
@@ -139,24 +138,18 @@ func (m *Message) WriteTo(w io.Writer) (int64, error) {
 	return total, nil
 }
 
-// DecodeMessage deseralizes data (as []byte) and creates a new Message
+// DecodeMessage deserializes data (as []byte) and creates a new Message
 func DecodeMessage(b []byte) (*Message, error) {
 	var msg Message
 
-	msg.Timestamp = int64(binary.BigEndian.Uint64(b[:8]))
-	msg.Attempts = binary.BigEndian.Uint16(b[8:10])
-
-	buf := bytes.NewBuffer(b[10:])
-
-	_, err := io.ReadFull(buf, msg.ID[:])
-	if err != nil {
-		return nil, err
+	if len(b) < 11+MsgIDLength {
```

Comment:
It's only confusing being 11 because nsqd rejects length 0 bodies. I'd expect in the code for this to be 10, the `len(int64)` and `len(uint16)`.

FWIW, NewMessage accepts length 0 bodies; it'd be more consistent for this to accept length 0 bodies. Also might make sense for nsqd to do the same. IIRC there was an issue/discussion about that somewhere.